File: MATLAB/NewtonRaphsonMet.m
function [mNR]=NewtonRaphsonMet(x0,f,tol,iterMax)
  % Declaracion de variables
  iter = 1:iterMax;                           % Definir iter como un vector
  df = diff(f);                               % Obtener funcion derivada de f
  err = tol + 1e0;                            % Error inicia sobre tol
  % Ciclo de iteraciones
  for i = iter
    if (subs(df,x0) == 0)                     % Verificar si hay divergencia
      disp("Diverge");
      break
    elseif (vpa(err) <= vpa(tol))             % Verificar si se llego a tol
      disp("Maxima tolerancia alcanzada");
      break      
    else                                      % Proceder a iteracion
      xi = vpa(x0 - subs(f,x0)/subs(df,x0));  % Calcular nueva x
      err = abs(x0 - xi);                     % Obtener error absoluto
      x0 = xi;                                % Redefinir x0 como xi
      mNR(i,1) = fix(vpa(i));
      mNR(i,2) = xi;                          % Almacenar en mNR
      mNR(i,3) = err;
    end
  end
end

File: Python/NewtonRaphsonMet.py
import sympy as sym

def newton_raphson_met(x0,f,tol,iter_max):
    # Variable declaration
    m_nr = []
    x = sym.Symbol('x')
    iter = range(iter_max)  # List for iteration
    df = sym.diff(f)        # Derivative of f
    err = tol + 1           # Error should start greater than tolerance
    # Iteration cycle
    for i in iter:
        if df.subs(x,x0) == 0:
            print("Detected divergence")
            break
        elif err <= tol:
            print("Tolerance limit reached")
            break
        else:
            xi = x0 - f.subs(x,x0)/df.subs(x,x0) # New X
            err = abs(x0 - xi)                   # New error
            x0 = xi                              # Redefine X0
            m_nr.append([i,float(xi),float(err)])
    return(m_nr)
